### PR TITLE
[Consensus] cleanup after sequential event processor - part 3

### PR DESCRIPTION
**Pull request by zekun000**
## Motivation

With the sequential event processor we implemented, this PR we simplify the pacemaker by

- remove the Inner and Rwlock
- change timeout mechanism to a single channel

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

Existing tests.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

File: consensus/src/chained_bft/liveness/local_pacemaker.rs [hunk 1]
```diff
@@ -167,22 +163,24 @@ impl LocalPacemakerInner {
...
-            external_timeout_sender,
+            timeout_sender,
             pacemaker_timeout_manager: PacemakerTimeoutManager::new(
                 pacemaker_timeout_quorum_size,
                 highest_timeout_certificates,
                 persistent_liveness_storage,
             ),
-        }
+        };
+        // To jump start the execution return the new round event for the current round.
```
Comment (dmitri-perelman) [inline consensus/src/chained_bft/liveness/local_pacemaker.rs]: One day we'll get rid of bounded channels in places like this one and it'll be beautiful.
Btw this comment is obsolete, we're not returning anything here.

**Comment by zekun000** [comment]
@dmitri-perelman
Before the message goes to pacemaker then re-direct to event processor through another channel, now it's more clear how the message being processed imo(process_local_timeout).

Why does it feel safer to have a separate channel for the same timeout? The only difference when main loop stuck is we'll get some log messages until the extra channel becomes full and nothing else.

Our goal should  be making sure main loop always have proper timeout for each message.

**Comment by dmitri-perelman** [comment]
Imo a clean version of a Pacemaker trait should notify EventProcessor about new rounds and timeouts, but should not depend on EventProcessor main loop.
I find it slightly weird that we now have a `process_local_timeout()` trait method, which is invoked by EventProcessor in response to Pacemaker generating a timeout. The whole purpose of `process_local_timeout()` is to setup a new timeout, do you think we could keep it inside the Pacemaker implementation (upon timeout send an event and setup a next timeout)?

**Comment by andll** [comment]
FWIW I like having pacemaker work in main loop because we get rid of yet another thread/task/synchronization primitive. I think this alone is a big win.
Given that pacemaker data structure is non-concurrent and we need to grab a lock to work with it, I don't think there is benefit of having it in separate thread like we had before

**Comment by zekun000** [comment]
Replace Pacemaker trait with concrete struct. cc: @aching

**Pull request merged**